### PR TITLE
[Dynamo] add flex attention mode test

### PR DESCRIPTION
**Pull request by mlazos**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137121
* #137119
* #137227
* #137120
* #137117
* #137116
* #137115
* #137114

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Review by yanboliang — APPROVED** [review]

**Comment by laithsakka** [comment]
this stack of diffs regress the following benchmarks
wonder why it was not blocked by the test ! have to check the logs
```
{   "add_loop_eager": 5.6250449423949,   "add_loop_eager_dynamic": 2.7871885539129 }

```

**Comment by laithsakka** [comment]
ah i see this was intentional here
https://github.com/pytorch/pytorch/pull/137117

File: test/dynamo/test_modes.py [hunk 1]
```diff
@@ -581,6 +582,23 @@ def run_checks(setups_and_oplists, skips, ref_map):
...
+
+        flex_attention = torch.compile(flex_attention, dynamic=False)
+
+        prefix_lengths = torch.arange(8)
+
+        def prefix_lm(b, h, q, kv):
+            return prefix_lengths[b] >= kv
+
+        # This runs in fullgraph already
+        mask = create_block_mask(prefix_lm, 8, None, 512, 512, _compile=True)
```
Comment (Chillee) [inline test/dynamo/test_modes.py:600]: I think we should add `torch.compile(fullgraph=True)` for this test.

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**